### ISSUE TITLE
Slurp whitespace at the end of template

### ISSUE THREAD

**Issue by memeplex** [kevinic/texthon#4]
The current `$>` slurp syntax is handy but it has some shortcommings:

1. Usually, it's more convenient to strip trailing whitespace by default at the end of a template and to let the call site set the correct combination of whitespace and newlines.
2. When a template ends with a compound directive there is no way to strip trailing whitespace, as `$>` won't work after `#}`. This can be done at the call site, but this is not possible for `main` as it's never explicitly called.

Stripping trailing whitespace by default won't be backward compatible, so I propose two simple changes:

1. To add a configuration option to strip template trailing whitespace by default.
2. To allow `$>` after `#}`.

**Comment by kevinic** [comment]
For 1, I can possibly add an extra parameter to load, something like:
`#load "lib/file.tmpl" as lib (autostrip)`  In the current state, you can do something like `${lib.template_func().rstrip()}`, which seems reasonable to me as well.

For 2, $> shouldn't be necessary after any statement, including #}, since the entire line is consumed.  $> should go to the last text line.  if that's not the case, can you give me a test case?  I can then debug against that.

**Comment by memeplex** [comment]
I didn't realize about rstrip, I wasn't thinking about the result of the template as a plain old string. I still think stripping is a better default though. Also there is the problem with main.

Regarding the second point, I understand your observation, but it's inconvenient to strip just the last output of a loop or every branch of a conditonal.

**Comment by kevinic** [comment]
Yeah, I think having a global strip option is reasonable.  One of the goals of Texthon is to help with readability, and having $> everywhere isn't that great.  Let me see what I can do.

If I take care of that, does that address your issue with main and compound statements?  I don't think I fully understand your issue with compound statements.  It'd be really helpful if you can give me a template snippet and tell me what your expected output is.  Thanks!
